scripts/env/perlmutter.sh

```diff
@@ -0,0 +1,14 @@
+#!/bin/sh -e
+
+# The cudatoolkit on Perlmutter is provided by NVHPC, which moved some CUDA headers to a "math_libs" directory (e.g., curand_kernel.h).
+# If using Spack, this will cause a compile error in VecGeom+cuda.
+# Spack uses this CUDA install as external CUDA package and doesn't look in the math_libs directory for extra headers.
+# --> the fix for now is to unload these modules and install our own cudatoolkit using Spack.
+module unload gpu cudatoolkit
+module load PrgEnv-gnu/8.3.3
+
+# Spack module on Perlmutter currently fails to create the spack env from spack.yaml, so we use our own install
+_SPACK_INSTALL=${SCRATCH}/spack
```

Comment:
Is `SCRATCH` defined particular to your user environment? It might be good to add a comment and assert that it's defined.

Reply:
Yes it's user-specific but defined by default on Perlmutter. I'll add a check that the file `setup-env.sh` exists